src/dump.c

```diff
@@ -286,10 +287,10 @@ static size_t queue_external_mis(jl_array_t *list)
 {
     size_t i, n = 0;
     htable_t visited;
-    htable_new(&visited, 0);
     if (list) {
         assert(jl_is_array(list));
         size_t n0 = jl_array_len(list);
+        htable_new(&visited, n0);
```

Comment:
The `htable_free` below now access uninitialized memory and must be moved also

Reply:
Thanks! Fixed in #44446